### PR TITLE
Add a command to invert conditions to the event sheet context menu

### PR DESCRIPTION
**Pull request by blurymind**
It inverts all conditions inside a Selection. If the selection has no conditions, dont show the command in the menu (new function to check for that in the selection handler)

discussion here:
https://github.com/4ian/GDevelop/issues/685

Demo:
invertconditionsgd

### CODE REVIEW

File: newIDE/app/src/EventsSheet/SelectionHandler.js [hunk 1]
```diff
@@ -105,6 +105,19 @@ export const hasInstructionSelected = (selection: SelectionState): boolean => {
   return !!Object.keys(selection.selectedInstructions).length;
 };
 
+export const hasSelectedAtleastOneCondition = (
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/SelectionHandler.js]: nitpicking: `hasSelectedAtLeastOneCondition` with a capital L ;)

File: newIDE/app/src/EventsSheet/SelectionHandler.js [hunk 2]
```diff
@@ -105,6 +105,19 @@ export const hasInstructionSelected = (selection: SelectionState): boolean => {
...
+  selection: SelectionState
+): boolean => {
+  let hasCondition = false;
+  for (var instructionContext of getSelectedInstructionsContexts(selection)) {
+    if (instructionContext.isCondition) {
+      hasCondition = true;
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/SelectionHandler.js]: nitpicking: You can remove hasCondition boolean by returning true here...

File: newIDE/app/src/EventsSheet/SelectionHandler.js [hunk 3]
```diff
@@ -105,6 +105,19 @@ export const hasInstructionSelected = (selection: SelectionState): boolean => {
...
+      break;
+    }
+  }
+  return hasCondition;
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/SelectionHandler.js]: ...and return false here

File: newIDE/app/src/EventsSheet/SelectionHandler.js [hunk 4]
```diff
@@ -105,6 +105,19 @@ export const hasInstructionSelected = (selection: SelectionState): boolean => {
   return !!Object.keys(selection.selectedInstructions).length;
 };
 
+export const hasSelectedAtleastOneCondition = (
+  selection: SelectionState
+): boolean => {
+  let hasCondition = false;
+  for (var instructionContext of getSelectedInstructionsContexts(selection)) {
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/SelectionHandler.js]: Use `let` instead of var

**Comment by 4ian** [comment]
Added a comments about a few minor things. Otherwise looks good! Thanks for doing this :)

**Comment by blurymind** [comment]
Thank you for reviewing. Good catches too :)

**Comment by blurymind** [comment]
ah strange, removing the boolean with true/false has somehow broken the flow test

File: newIDE/app/src/EventsSheet/SelectionHandler.js [hunk 5]
```diff
@@ -105,6 +105,18 @@ export const hasInstructionSelected = (selection: SelectionState): boolean => {
   return !!Object.keys(selection.selectedInstructions).length;
 };
 
+export const hasSelectedAtLeastOneCondition = (
+  selection: SelectionState
+): boolean => {
+  for (let instructionContext of getSelectedInstructionsContexts(selection)) {
+    if (instructionContext.isCondition) {
+      return true;
+      break;
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/SelectionHandler.js]: You don't need a break after a return, return will end the current function ;)

Comment (blurymind) [reply newIDE/app/src/EventsSheet/SelectionHandler.js]: oooh *slaps forehead* xD

Comment (blurymind) [reply newIDE/app/src/EventsSheet/SelectionHandler.js]: sorry, late night coding silliness again

Comment (4ian) [reply newIDE/app/src/EventsSheet/SelectionHandler.js]: Flow was good enough this time to let you know about this ;)
image

**Comment by 4ian** [comment]
Looks good! Thanks for the PR :)

**Comment by blurymind** [comment]
Thank you too! 

good night :)

**Comment by 4ian** [comment]
Thanks, you too ;)

**Pull request merged**